pkgs/applications/misc/rtfm/default.nix

```diff
@@ -0,0 +1,66 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, fetchurl
+, callPackage
+, crystal_1_8
+, libadwaita
+, wrapGAppsHook4
+, gobject-introspection
+, desktopToDarwinBundle
+, webkitgtk_6_0
+, sqlite
+, gi-crystal
+}:
+let
+  gtk-docs = callPackage ./gtk-docs.nix { };
+in
+crystal_1_8.buildCrystalPackage rec {
+  pname = "rtfm";
+  version = "0.2.1";
+
+  src = fetchFromGitHub {
+    owner = "hugopl";
+    repo = "rtfm";
+    rev = "v${version}";
+    name = "rtfm";
+    hash = "sha256-ZhKwnfoU3LgCniJn3Ino1MlHfmPQCyz9LLt+7ci9iBk=";
+  };
+
+  crystal_doc_version = "1.9.0";
+
+  crystal-docs = fetchurl {
+    url = "https://github.com/crystal-lang/crystal/releases/download/${crystal_doc_version}/crystal-${crystal_doc_version}-docs.tar.gz";
+    hash = "sha256-OC36durhxEh/OYztjS4r9JSqpqL8f8At7+bFXQQrzhM=";
+  };
+
+  patches = [
+    ./make.patch
```

Comment:
Could you please add a comment on why this patch is required, for each patch in this list?

Reply:
Sure